provider/spark_test.go

```diff
@@ -2880,3 +2883,428 @@ func createEMRAndS3(bucketName string) (SparkExecutor, SparkFileStore, error) {
+		},
+	}
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			pythonURI := tt.executor.PythonFileURI(store)
+			if !reflect.DeepEqual(pythonURI, tt.ExpectedPythonFileURI) {
+				t.Errorf("SparkExecutor.PythonFileURI() = %#v, want %#v", pythonURI, tt.ExpectedPythonFileURI)
+			}
+			submitArgs := tt.executor.SparkSubmitArgs(tt.SubmitArgs.DestPath, tt.SubmitArgs.Query, tt.SubmitArgs.SourceList, tt.SubmitArgs.JobType, store)
+			if !reflect.DeepEqual(submitArgs, tt.ExpectedSubmitArgs) {
+				t.Errorf("SparkExecutor.SubmitArgs() = %#v, want %#v", submitArgs, tt.ExpectedSubmitArgs)
+			}
+			dfArgs, err := tt.executor.GetDFArgs(tt.DFArgs.OutputURI, tt.DFArgs.Code, tt.DFArgs.Sources, store)
+			if err != nil {
+				t.Errorf("SparkExecutor.GetDFArgs() = %#v, want %#v", err, nil)
+			}
+			if !reflect.DeepEqual(dfArgs, tt.ExpectedDFArgs) {
+				t.Errorf("SparkExecutor.GetDFArgs() = %#v, want %#v", dfArgs, tt.ExpectedDFArgs)
+			}
+		})
+	}
+}
+
+func TestSparkOfflineStore_getResourceInformationFromFilePath(t *testing.T) {
+	type fields struct {
+		Executor     SparkExecutor
+		Store        SparkFileStore
+		Logger       *zap.SugaredLogger
+		query        *defaultPythonOfflineQueries
+		BaseProvider BaseProvider
+	}
+	type args struct {
+		path string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   string
+		want1  string
+		want2  string
+	}{
+		{
+			"Short S3",
+			fields{},
+			args{"s3://bucket/path/to/file"},
+			"",
+			"",
+			"",
+		},
+		{
+			"Long S3",
+			fields{},
+			args{"s3://bucket/long/path/to/file"},
+			"path",
+			"to",
+			"file",
+		},
+		{
+			"Short S3a",
+			fields{},
+			args{"s3a://bucket/path/to/file"},
+			"",
+			"",
+			"",
+		},
+		{
+			"Long S3a",
+			fields{},
+			args{"s3a://bucket/long/path/to/file"},
+			"path",
+			"to",
+			"file",
+		},
+		{
+			"Short hdfs",
+			fields{},
+			args{"hdfs://bucket/path/to/file"},
+			"",
+			"",
+			"",
+		},
+		{
+			"Long hdfs",
+			fields{},
+			args{"hdfs://bucket/long/path/to/file"},
+			"path",
+			"to",
+			"file",
+		},
+		{
+			"Featureform Path Short",
+			fields{},
+			args{"featureform/something"},
+			"",
+			"",
+			"",
+		},
+		{
+			"Featureform Path Long",
+			fields{},
+			args{"featureform/long/path/to/file"},
+			"long",
+			"path",
+			"to",
+		},
+		{
+			"Featureform__ Path Short",
+			fields{},
+			args{"featureform_"},
+			"",
+			"",
+			"",
+		},
+		{
+			"Featureform__ Path Long",
+			fields{},
+			args{"featureform__long__path__to"},
+			"_long",
+			"path",
+			"to",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			spark := &SparkOfflineStore{
+				Executor:     tt.fields.Executor,
+				Store:        tt.fields.Store,
+				Logger:       tt.fields.Logger,
+				query:        tt.fields.query,
+				BaseProvider: tt.fields.BaseProvider,
+			}
+			got, got1, got2 := spark.getResourceInformationFromFilePath(tt.args.path)
```

Comment:
I'd match these identifiers to the return names. 
`fileType, fileName, fileVariant`